sdk-python/list_and_match_messages.py

```diff
@@ -0,0 +1,72 @@
+import argparse
+from datetime import datetime
+
+from awaazde import AwaazDeAPI
+from awaazde.constants import CommonConstants, APIConstants
+from awaazde.utils import CSVUtils
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Check Created Messages')
+    parser.add_argument('username', type=str, help='Username of the tenant')
+    parser.add_argument('password', type=str, help='Password of the tenant')
+    parser.add_argument('organization', type=str, help='Organization of the tenant')
+    parser.add_argument('match_path', type=str, help='Path for the csv with which filtered data is to be matched')
+    parser.add_argument("--params", nargs='+', help="Fields on which filters are to be applied."
+                                                    "Set a number of key-value pairs within double-quotes "
+                                                    "(do not put spaces before or after the = sign)."
+                                                    "Example: --params 'send_on__gt'='10-12-2020' 'tags'='dummy_tag' ")
+    parser.add_argument('--match_criteria', type=str,
+                        help='Field to use when matching User record with the queried resultant record',
+                        default=CommonConstants.MESSAGE_ID)
+    return parser.parse_args()
+
+
+def parse_vars(extra_vars):
+    """
+    Take a list of comma separated key value pair strings, separated
+    by comma strings like 'foo=bar' and return as dict.
+    :param extra_vars: list[str] ['foo=bar, 'key2=value2']
```

Comment:
This funct name and param name aren't fitting. Why are these vars? What is extra about the vars? This func takes a key-val string and converts into a dict. BTW there's got to be a more Pythonic way to do it than what you've got here. Probably a 1-2 liner, in which case you probably don't need a separate func at all. Make sure you reference whatever approach you use. If you keep it as a func, consider moving to to common utils